EIPS/eip-7484.md

```diff
@@ -0,0 +1,217 @@
+
+### Adapter
+
+In order for smart accounts to increase security guarantees when adding modules, they must be able to securely query the Module Registry and handle the return data correctly. In order to achieve this, this proposal aims to provide a standardised interface that may be implemented by smart accounts irrespective of their architecture, execution flows and security assumptions.
+
+### Related work
+
+The reference implementation of the Registry is heavily inspired by the Ethereum Attestation Service. The specific use-case of this proposal, however, required some custom modifications and additions to EAS, meaning that using the existing EAS contracts as the Module Registry was sub-optimal.
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Reference Implementation
+
+### Adapter.sol
+
+```solidity
+contract Adapter {
+    IRegistry registry;
+
+    function checkModule(address module, address trustedAttester) internal {
+       // check module implementation address on registry
+        (uint48 listedAt, uint48 flaggedAt) = registry.check(module, trustedAttester);
+
+        // revert if module was ever flagged or was never attested to
+        require(listedAt != 0 && flaggedAt == 0, "Module is insecure");
+    }
+
+    function verifyModule(address module, address[] memory attesters, uint256 threshold) internal {
+        // check module implementation address on registry
+        bool verified = registry.verify(module, attesters, threshold);
+
+        // revert if module is not verified by enough attesters
+        require(verified, "Module is insecure");
+    }
+}
+```
+
+### Account.sol
+
+**Note**: This is a specific example that complies to the `Specification` above, but this implementation is not binding.
+
+```solidity
+contract Account is Adapter {
+    ...
+
+    // installs a module
+    function installModule(address module, address trustedAttester) public {
+       checkModule(module, trustedAttester);
+        ...
+    }
+
+    // executes a module
+    function executeTransactionFromModule(address module, address[] memory attesters, uint256 threshold) public {
+        verifyModule(module, attesters, threshold);
+        ...
+    }
+
+    ...
+}
+```
+
+### Registry
+
+See `https://github.com/rhinestonewtf/registry/`
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
If you use an HTML style comment, the linter will make sure you replace it before you go to review:

```suggestion
Needs discussion. <!-- TODO -->
```